### PR TITLE
Fix a number of memory leaks.

### PR DESCRIPTION
**Pull request by susilehtola**
## Description
This commit fixes a number of memory leaks revealed by cppcheck.

## Status
- [x]  Ready to go

### CODE REVIEW

File: psi4/src/psi4/ccdensity/file_build.cc [hunk 1]
```diff
@@ -102,18 +102,33 @@ int file_build(dpdfile4 *File, int inputfile, double tolerance,
               core_left = memoryd - row_length;
 
               /* Make room for another bucket */
-              bucket_offset = (int **) realloc((void *) bucket_offset,
-                                               nbuckets * sizeof(int *));
```
Comment (dgasmith) [inline psi4/src/psi4/ccdensity/file_build.cc]: We are a bit hit or miss in catching `alloc`, `realloc`, `calloc`, etc calls. However, this one is particularly large so probably a good idea to do so. For each of these can you include explicit bracers in the if/else calls.

File: psi4/src/psi4/dcft/dcft_sort_mo_tpdm.cc [hunk 2]
```diff
@@ -92,18 +92,35 @@ void DCFTSolver::presort_mo_tpdm_AB()
...
                 coreLeft = memoryd - rowLength;
                 /* Make room for another bucket */
-                bucketOffset = (int **) realloc((void *) bucketOffset,
-                                             nBuckets * sizeof(int *));
+
+		int **p;
+
+		p = static_cast<int **>(realloc(static_cast<void *>(bucketOffset),
+						nBuckets * sizeof(int *)));
+		if(p == NULL)
```
Comment (amjames) [inline psi4/src/psi4/dcft/dcft_sort_mo_tpdm.cc]: Should probably get explicit braces in here as well?

File: psi4/src/psi4/dcft/dcft_sort_mo_tpdm.cc [hunk 3]
```diff
@@ -230,18 +247,35 @@ void DCFTSolver::presort_mo_tpdm_AA()
...
```
Comment (amjames) [inline psi4/src/psi4/dcft/dcft_sort_mo_tpdm.cc]: Braces in here as well, improves readability.

**Comment by amjames** [comment]
@susilehtola Thanks! this branch just needs a rebase then I think it's ready to go!

**Pull request merged**